### DISCUSSION TITLE
[Showcase & Feedback] Using YOLOv8 for Automated Feature Extraction in Psychological House-Tree-Person (HTP) Sketches

### DISCUSSION THREAD

**Discussion by kx-naux** [ultralytics/ultralytics#25495] [Solutions]
Hi Ultralytics Community! 👋

I am a student research team from **Tunku Abdul Rahman University of Management and Technology (Penang Branch, Malaysia)** developing **MindLens**—a digital health ecosystem designed to assist in early mental health risk stratification. 

To overcome self-report bias (like "faking good" on standard forms), we engineered a **Hybrid Assessment Model** that pairs standardized metrics (DASS-42) with an automated **House-Tree-Person (HTP) Projective Test**.

---

📁 **Attachment:** You can check out our complete [MindLens Concept Deck (PDF)](https://github.com/user-attachments/files/30494866/MindLens.Idea.pdf) for an overview of the full project ecosystem!

### 🛠️ Our Machine Learning & Vision Architecture

Our pipeline relies on YOLOv8 to perform bounding-box object detection and segmentation on hand-drawn user sketches:

1. **Preprocessing:** Image restoration and clean-up using OpenCV.
2. **Detection Layer (YOLOv8):** Bounding box detection to locate and isolate 3 core classes: `House`, `Tree`, and `Person`.
3. **Spatial Feature Extraction:** Extracting spatial geometry metrics—such as relative scale ratios, canvas positioning percentage, bounding-box overlapping, and line quality.
4. **Downstream Classification:** Passing extracted spatial vectors into a Random Forest classifier alongside DASS-42 scores for non-diagnostic risk stratification.

---

### ❓ What We’d Love Your Feedback On

Hand-drawn sketches behave quite differently from natural scene images (photographs). We would be super grateful for advice or insights from the Ultralytics team, ambassadors, and community members on:

1. **Hyperparameter Tuning for Sketches / Line Art:**
   * What data augmentation techniques work best for minimalist black-and-white drawings (e.g., scale, rotation, mosaic vs. shear)?
   * Recommendations on optimizing anchor box priors or loss gains (e.g., `box`, `cls`) when dealing with sparse stroke data?

2. **Handling Occlusion & Overlap:**
   * In HTP sketches, subjects frequently draw the `Person` overlapping with or inside the `House` or under a `Tree`. Any tips on improving bounding-box recall for overlapping/nested elements?

3. **General Feedback:**
   * We’d love any general thoughts, advice, or suggestions on our architectural pipeline!

Thank you so much to the Ultralytics team for creating such an incredible, accessible framework!

**Low Kah Xuan**  
🔗 LinkedIn: linkedin.com/in/low-kah-xuan

**Comment by UltralyticsAssistant** [comment]
👋 Hello @kx-naux, thank you for sharing your MindLens project and your work with YOLOv8 for automated feature extraction in psychological House-Tree-Person sketches 🚀! This is an interesting application of computer vision to structured hand-drawn imagery.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users, where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many common questions may already be answered.

For additional feedback from the community, please include details such as dataset size and balance, annotation examples, training configuration, validation metrics, and representative prediction results. If you encounter a 🐛 bug, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) to help us debug it.

For custom training ❓ questions, please provide as much information as possible, including dataset image examples and training logs, and verify that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

This is an automated response 🤖, and an Ultralytics engineer will also assist you soon.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments, with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Thanks for sharing this with the community. YOLOv8 is anchor-free, so there are no anchor priors to tune; start with modest rotation, translation, and scale, disable color-heavy transforms, and compare mosaic on/off through controlled validation before adjusting `box` or `cls`. For nested or overlapping drawings, annotate every instance consistently, oversample overlap-heavy cases, consider higher-resolution training, and use [instance segmentation](https://docs.ultralytics.com/tasks/segment) when precise geometry matters more than rectangular extent. Keep sketch-derived features separate from clinical conclusions, validate on unseen participants and demographic subgroups, and retain qualified human oversight for any risk-stratification workflow.

**Comment by kx-naux** [comment]
Re: YOLOv8 for HTP Projective Drawing Analysis (MindLens)

Thank you so much for the detailed guidance on anchor-free tuning, augmentation strategies, and handling nested/overlapping elements! The advice on disabling color-heavy transforms, oversampling overlap-heavy cases, and considering instance segmentation for precise geometry is super actionable for our pipeline.

**Clinical & Human-in-the-Loop Validation:**
To address the point on clinical oversight: Yes, we currently have professional psychiatrists actively involved to validate our screening outcomes and overall diagnostic logic. We explicitly design MindLens so that sketch-derived features remain strictly a clinical triage aid, leaving final risk-stratification and clinical conclusions to qualified human professionals.

**Questions on Technical Feasibility & Computer Vision Strategy:**
While our clinical partners are validating the medical outcomes, we would love to get your thoughts on two specific computer vision challenges we are working through for sketch/line-art analysis:

1. Line Weight & Pressure Detection:
In HTP drawing analysis, stroke pressure (heavy/faint lines) is a significant clinical indicator. Since traditional object detection treats line art as binary/flat masks, what is the best recommended approach in a YOLO/OpenCV pipeline to quantify line weight or stroke density? Would grayscale edge-intensity profiling or HSV-luminance mapping within bounding boxes be effective prior to feeding features into our classifier?

2. Corner Sharpness & Angularity:
The sharpness vs. roundness of drawn corners (e.g., sharp angular roof edges vs. curved rounded edges in the "House" test) carries diagnostic weight. Is instance segmentation (YOLOv8-seg) sufficient to extract contour curvature/polygonal approximation (e.g., via cv2.approxPolyDP on predicted masks), or would a dedicated secondary feature extraction step be cleaner?

**Technical Endorsement & Collaboration:**
Since we are showcasing MindLens at an upcoming hackathon, we want to ensure our computer vision architecture follows state-of-the-art best practices.

If our pipeline design looks solid to the Ultralytics team/community, would you be open to providing a brief technical quote or endorsement regarding our application of YOLOv8 for digital health / line-art feature extraction? Having technical feedback from the creators and experts behind YOLO would mean a lot to our team and add huge credibility to our pitch!

Looking forward to your thoughts!

**Reply by glenn-jocher** [reply]
For line weight, preserve the original grayscale image and calibrated capture conditions, then compute intensity percentiles, foreground density, and estimated stroke width within each predicted region; HSV adds little for monochrome sketches, and image darkness should not be treated as physical pressure without validation. For angularity, use [instance segmentation](https://docs.ultralytics.com/tasks/segment) to isolate each object, then run a dedicated OpenCV feature stage on cleaned masks using contours, curvature, and `cv2.approxPolyDP`, testing sensitivity to resolution and smoothing. This is a reasonable research pipeline when features are validated independently and retained as non-diagnostic signals, but we’re unable to provide a formal technical endorsement or promotional quote.
